extensions-core/stats/src/main/java/io/druid/query/aggregation/teststats/ZtestPostAggregator.java

```diff
@@ -0,0 +1,159 @@
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Sets;
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.query.aggregation.PostAggregator;
+import io.druid.query.aggregation.post.ArithmeticPostAggregator;
+import io.druid.query.aggregation.post.PostAggregatorIds;
+import io.druid.query.cache.CacheKeyBuilder;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/*
+   * 1. calculating zscore using two-sample Z-Test. IOW,
+   *    using z-test statistic for testing the difference of
+   *    two population proportions.
+   * 2. converting binary variables (e.g. success or not) to continuous variables (e.g. conversion rate).
+   *
+   Please refer to http://math.mercyhurst.edu/~griff/courses/m109/Lectures/old/Sum_06/sect8.1.pdf
+   for more details.
+   http://facweb.cs.depaul.edu/sjost/csc423/documents/test-descriptions/indep-z.pdf
+*/
+@JsonTypeName("zscore2sample")
+public class ZtestPostAggregator implements PostAggregator {
+  private final String name;
+  private final List<PostAggregator> fields;
+
+  @JsonCreator
+  public ZtestPostAggregator(
+      @JsonProperty("name") String name,
+      @JsonProperty("fields") List<PostAggregator> fields
+  ) {
+    Preconditions.checkNotNull(name, "Must have a valid, non-null post-aggregator name");
+    Preconditions.checkArgument(fields.size() == 4, "Must have 4 fields; " +
+        "\"fields\": <total success count of sample 1>, <sample size of population 1>," +
+        " <total success counts of sample 2>, <sample size of population 2>");
+    this.name = name;
+    this.fields = fields;
+  }
+
+  @Override
+  public Set<String> getDependentFields() {
+    Set<String> dependentFields = Sets.newLinkedHashSet();
+    for (PostAggregator field : fields) {
+      dependentFields.addAll(field.getDependentFields());
+    }
+    return dependentFields;
+  }
+
+  @Override
+  public Comparator getComparator() {
+    return ArithmeticPostAggregator.DEFAULT_COMPARATOR;
+  }
+
+  @Override
+  public Object compute(Map<String, Object> combinedAggregators) {
+
+    PostAggregator[] args = new PostAggregator[fields.size()];
+    for (int j = 0; j < fields.size(); j++) {
+      args[j] = fields.get(j);
+    }
+
+    return zScoreTwoSamples(
+        ((Number) args[0].compute(combinedAggregators)).doubleValue(),
+        ((Number) args[1].compute(combinedAggregators)).doubleValue(),
+        ((Number) args[2].compute(combinedAggregators)).doubleValue(),
+        ((Number) args[3].compute(combinedAggregators)).doubleValue());
+  }
+
+  @Override
+  @JsonProperty
+  public String getName() {
+    return name;
+  }
+
+  @Override
+  public PostAggregator decorate(Map<String, AggregatorFactory> aggregators) {
+    return this;
```

Comment:
Like PvaluefromZscorePostAggregator, this should recursively decorate the post-aggregators from `fields`.